src/BackgroundLocationTracking/MainPage.xaml.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Esri.ArcGISRuntime.Location;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Controls;
+using System;
+using System.Threading.Tasks;
+using Location = Esri.ArcGISRuntime.Location.Location;
+
+namespace BackgroundLocationTracking
+{
+    public partial class MainPage : ContentPage
+    {
+        private LocationDataSource? _locationDataSource;
+
+        public MainPage()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            try
+            {
+#if IOS
+                _locationDataSource = new SystemLocationDataSource
+                {
+                    AllowsBackgroundLocationUpdates = true
+                };
+#else
+                _locationDataSource = new SystemLocationDataSource();
+#endif
+            }
+            catch (Exception ex)
+            {
+                // Handle initialization errors
+                LogMessage.Text = $"Error: An error occurred during initialization: {ex.Message}";
+            }
+        }
+
+        /// <summary>
+        /// Starts the location service.
+        /// </summary>
+        private async void StartTracking(object sender, EventArgs e)
+        {
+            if (_locationDataSource is null ||
+                await CheckAndRequestLocationPermission() is not PermissionStatus.Granted)
+            {
+                return; // Exit if the location data source is not initialized.
+            }
+#if ANDROID
+                var intent = new Android.Content.Intent(Android.App.Application.Context, typeof(LocationService));
+                // Check if the Android version is at least Oreo (API level 26)
+                // Start the service as a foreground service
+                _ = Android.App.Application.Context.StartForegroundService(intent);
+#endif
+            await StartLocationDataSource();
+        }
+
+        private async Task StartLocationDataSource()
+        {
+            if (_locationDataSource is not null)
+            {
+                // Subscribe to the LocationChanged event
+                _locationDataSource.LocationChanged += LocationDataSource_LocationChanged;
```

Comment:
Do this on datasource creation instead, so you don't have to subscribe and unsubscribe every time